hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/ConcentricCylindersGeometry.h
+ * \brief Definition of the MPCD slit channel geometry
+ */
+
+#ifndef MPCD_CONCENTRIC_CYLINDERS_GEOMETRY_H_
+#define MPCD_CONCENTRIC_CYLINDERS_GEOMETRY_H_
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+
+#ifdef __HIPCC__
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // __HIPCC__
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Concentric cylinders geometry
+
+class __attribute__((visibility("default"))) ConcentricCylindersGeometry
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param R0 Inner radius of the cylinder
+     * \param R1 Outer radius of the cylinder
+     * \param speed Speed of the outer cylinder
+     * \param no_slip Boundary condition at the wall (slip or no-slip)
+     */
+    HOSTDEVICE ConcentricCylindersGeometry(Scalar R0, Scalar R1, Scalar speed, bool no_slip)
+        : m_R0(R0), m_R02(R0 * R0), m_R1(R1), m_R12(R1 * R1), m_V(speed), m_no_slip(no_slip)
+        {
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar r = pos.x * pos.x + pos.y * pos.y;
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((r < m_R02) - (r > m_R12));
```

Comment:
Similarly here, you might find it a little easier to read the variables if you add `sq` or `_sq`
```suggestion
        const signed char sign = (char)((r < m_R0_sq) - (r > m_R1_sq));
```